tests/eth_fixtures.py

```diff
@@ -16,51 +17,83 @@
 
 
 @pytest.fixture(scope='session')
-def sol_compiler():
+def solidity_compiler():
     compiler = SolidityCompiler()
     yield compiler
 
 
 @pytest.fixture(scope='module')
-def tester_registrar():
+def registrar():
     _, filepath = tempfile.mkstemp()
-    tester_registrar = Registrar(chain_name='tester', registrar_filepath=filepath)
-    yield tester_registrar
+    registrar = Registrar(chain_name='tester', registrar_filepath=filepath)
+    yield registrar
     os.remove(filepath)
 
 
 @pytest.fixture(scope='module')
-def tester_provider(tester_registrar, sol_compiler):
+def geth_ipc_provider(registrar, solidity_compiler):
     """
     Provider backend
     https: // github.com / ethereum / eth - tester     # available-backends
     """
-    eth_tester = EthereumTester(backend=PyEVMBackend())
-    test_provider = EthereumTesterProvider(ethereum_tester=eth_tester)  # , api_endpoints=None)
+    #
+    # spin-up geth
+    #
+    testing_dir = tempfile.mkdtemp()
```

Comment:
Context manager? https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

Reply:
We decided against this for better multi-platform support.